OpenRA.Mods.Common/Traits/World/WeatherOverlay.cs

```diff
@@ -0,0 +1,421 @@
+		[Desc("Can be snow or rain or d2kSand")]
+		public readonly string WeatherVariants = "snow";
+		[Desc("Can be true or false")]
+		public readonly bool Wind = true;
+		[Desc("Can be true or false")]
+		public readonly bool InstantWind = false;
+		[Desc("Can be true or false")]
+		public readonly bool AutoWind = true;
+		[Desc("Works only when AutoWind is disabled. Can be: StormLeft, StrongLeft, WeakLeft, Calm, WeakRight, StrongRight, StormRight")]
+		public readonly WeatherOverlay.WindTypes WindType = WeatherOverlay.WindTypes.Calm;
+
+		public object Create(ActorInitializer init) { return new WeatherOverlay(init.World, this); }
+	}
+
+	public class WeatherOverlay : ITick, IRenderWeatherOverlay
+	{
+
+		readonly WeatherOverlayInfo info;
+		readonly World world;
+
+		public struct droppingParticle
+		{
+			public float PosX;
+			public float PosY;
+			public int   Size;
+			public float DirectionScatterX;
+			public float Gravity;
+			public float SwingOffset;
+			public float SwingSpeed;
+			public bool  SwingDirection;
+			public float SwingAmplitude;
+			public Color Color;
+			public Color TailColor;
+
+			public droppingParticle(float posX, float posY, int size, float directionScatterX, float gravity, float swingOffset, float swingSpeed, bool swingDirection, float swingAmplitude, Color color, Color tailColor)
+			{
+				PosX = posX;
+				PosY = posY;
+				Size = size;
+				DirectionScatterX = directionScatterX;
+				Gravity = gravity;
+				SwingOffset = swingOffset;
+				SwingSpeed = swingSpeed;
+				SwingDirection =  swingDirection;
+				SwingAmplitude = swingAmplitude;
+				Color = color;
+				TailColor = tailColor;
+			}
+		} 
+		public List<droppingParticle> DroppingParticleList = new List<droppingParticle>();
+		int MaxDrPaCount = CalculateDroppingParticleCount(Game.Renderer.Resolution.Width, Game.Renderer.Resolution.Height);
+
+		enum DrPaCountFaderType { Hold, FadeIn, FadeOut };
+		DrPaCountFaderType DrPaCountFader = DrPaCountFaderType.FadeIn;
+
+		float TargetWindXOffset = 0f;
+		float CurrentWindXOffset = 0f;
+		public enum WindTypes {
+			StormLeft,
+			StrongLeft,
+			WeakLeft,
+			Calm,
+			WeakRight,
+			StrongRight,
+			StormRight,
+		};
+		public WindTypes WindType = WindTypes.Calm;
+		long WindTickCountdown = 1500;
+		float2 AnitScrollPrevTopLeft;
+
+		public WeatherOverlay(World world, WeatherOverlayInfo info)
+		{
+			this.info = info;
+			this.world = world;
+		}
+
+		bool ChancesPercent(int percent)
+		{
+			if (Game.CosmeticRandom.Next(0, 101) <= percent)
+				return true;
+			else return false;
+		}
+
+		static int CalculateDroppingParticleCount(int x, int y )
+		{  
+			return (x * y) / 1311;  // ~ 800 particles at 1366x768px
+		}
+
+		// Adds DroppingParticles scattered over the Y-Axis range with a spawnchance of %
+		void AddADroppingParticle(int count, int rangeY, int spawnChancePercent)
+		{
+
+			for (int i = 0; i < count; i++)
+			{
+				if (ChancesPercent(spawnChancePercent))
```

Comment:
Please just be direct: `if (Game.CosmeticRandom.Next(0, 100) <= spawnChancePercent)` and then remove the unnecessary `ChancesPercent`.